include/glow/Importer/CommonOperatorLoader.h

```diff
@@ -857,6 +857,27 @@ class CommonOperatorLoader : public ProtobufLoader {
     return true;
   }
 
+  llvm::Expected<bool> loadGatherRanges(const std::string &typeName,
+                                        const OpType &op,
+                                        const ArgumentDictionaryTy &dict) {
+    NodeValue data;
+    ASSIGN_VALUE_OR_RETURN_ERR(data,
+                               getNodeValueOrCreateConstantByName(op.input(0)));
+
+    NodeValue ranges;
+    ASSIGN_VALUE_OR_RETURN_ERR(ranges,
+                               getNodeValueOrCreateConstantByName(op.input(1)));
+
+    unsigned_t maxOutputSize;
+    ASSIGN_VALUE_OR_RETURN_ERR(maxOutputSize,
+                               loadInt(dict.find("maxOutputSize")->second));
```

Comment:
where is 'maxOutputSize' coming from? I cannot see it in https://caffe2.ai/docs/operators-catalogue.html#gatherranges

Reply:
As @artemrakhov and I discussed in #2210, the shape of `output` can not be determined at compile time without a hint. `maxOutputSize` specifies the largest possible size of `output`, and any nodes that use it must use the value of `lengths` to determine how much of `output` is valid data (since `sum(lengths) == output.size()`).